universal-application-tool-0.0.1/app/controllers/admin/AdminQuestionController.java

```diff
@@ -356,14 +360,21 @@ private void updateDefaultLocalizationForOptions(
     // translations. If we do not have existing translations for a given string, create
     // a new, empty set of translations.
     ImmutableList.Builder<QuestionOption> updatedOptionsBuilder = ImmutableList.builder();
-    for (int i = 0; i < updatedDefaultOptions.size(); i++) {
-      String updatedDefaultOption = updatedDefaultOptions.get(i);
-      if (existingTranslations.containsKey(updatedDefaultOption)) {
-        updatedOptionsBuilder.add(
-            existingTranslations.get(updatedDefaultOption).toBuilder().setId(i).build());
+    for (QuestionOption updatedOption : updatedOptions) {
+      if (existingTranslations.containsKey(updatedOption.optionText().getDefault())
+          && existingTranslations.get(updatedOption.optionText().getDefault()).id()
+              == updatedOption.id()) {
+        QuestionOption existingOption =
+            existingTranslations.get(updatedOption.optionText().getDefault());
+        if (existingOption.id() == updatedOption.id()) {
```

Comment:
(blocking) is this true from line 365 or am I reading poorly?

Reply:
Nope, you're right, removed.